### PR TITLE
Fix unreviewed staleness timeline

### PR DESCRIPTION
**Pull request by elibarzilay**
Two changes:

* The `Unreviewed` timeline needs `!(requireMaintainer && !blessable)`
  to start.

* `noOtherOwners` => `requireMaintainer`.

Also:

* Reorganize the subterms in `blessable` to match `requireMaintainer` so
  the relationship between the two is clearer.

* Change `Comments.StalenessComment` so the index returns a
  function (instead of the other way around), which makes it possible to
  spit out an error line if it uses `otherOwners` and it is empty.

* Add this error to `comments.ts`.  Note that it doesn't throw an
  exception, just prints the error.  This is because if it happens, it's
  not a big problem to mention `«anyone?»`, but it is useful to debug
  such cases (which shouldn't happen, and with this fix they don't).

Fixes the new problem that was raised in #105.

### CODE REVIEW

File: src/compute-pr-actions.ts
Comment (jablko) [inline src/compute-pr-actions.ts:186]: It looks like this change (`noOtherOwners` => `requireMaintainer`) would change some PRs from `approverKind` "other" to `approverKind` "maintainer". What about the following instead?
```suggestion
        if (!approved && approverKind !== "maintainer") return mkStaleness(
```

Comment (elibarzilay) [reply src/compute-pr-actions.ts:186]: That's a good point: made me notice that I can simplify `getApproverKind` a lot with that change.

But I'm not following the purpose of the suggestion: that line has no effect on `approverKind`, so what's the purpose?  (Not objecting to the change since if a maintainer is required, then the `Unreviewed` timeline is just a cosmetic change since it'll be in the maintainer queue regardless, so I very slightly prefer my line just to minimize test (& behavior) changes.)

Comment (jablko) [reply src/compute-pr-actions.ts:186]: I wasn't clear, sorry. I'm suggesting this line instead of adding `noOtherOwners` to `requireMaintainer`, which *does* affect `approverKind` (specifically cases that were previously `approverKind` "other" and are now `approverKind` "maintainer").

Comment (elibarzilay) [reply src/compute-pr-actions.ts:186]: Ah, well -- the change in this case is *intended*...  Just in case, I looked inot it to verify that it works fine: the changed `approverKind` happens in two tests: 44282 and 49548 -- in both of these, there is no other owner, so it was bogus to have it get into a waiting-for-review in the first test, and get into the "unreviewed" stale state in the second.

Another way to see the problem is that with your suggestion, the `console.error` that I added to warn when there are no other owners *does* happen (for the unreviewed case, since it's trying to ping the other owners when there are none).

(It's true that there might be a point in waiting for a few days in a waiting-for-review state, in case someone steps in and provides a review, I'll ask about it.)

Comment (orta) [reply src/compute-pr-actions.ts:186]: I think generally making it auto switch is probably fine. If there was someone else who would give any sort of useful review in a reasonable timeframe I'd imagine they'd get added pretty soon?

Otherwise seems good to me 👍🏻

**Review by orta — APPROVED** [review]

File: src/compute-pr-actions.ts
Comment (jablko) [reply src/compute-pr-actions.ts:186]: When `approverKind` would be "owner" except that `noOtherOwners`, already that makes the `approverKind` "maintainer", which clearly it should be. +1 for fixing the unreviewed timeline in that case.

Changing `approverKind` "other" to `approverKind` "maintainer" feels like a different change. Not that I'm opposed, but if we do make that change, it should be uniform? It feels weird to me to make `approverKind` "maintainer" if there's one owner and "other" if there are multiple. Or "maintainer" if the owner made the change and "other" if someone else did?

**Pull request closed without merging**
